File: microtome/test/PrimitiveListPage.py
# GENERATED IMPORTS START
from microtome.core.prop import Prop
from microtome.core.prop import PropSpec
from microtome.page import Page
# GENERATED IMPORTS END

# GENERATED CLASS_DECL START
class PrimitiveListPage(Page):
# GENERATED CLASS_DECL END
# GENERATED CONSTRUCTOR START
    _s_inited = False
    def __init__(self, name):
        super(PrimitiveListPage, self).__init__(name)
        if not PrimitiveListPage._s_inited:
            PrimitiveListPage._s_inited = True
            PrimitiveListPage._s_stringsSpec = PropSpec("strings", None, [list, str, ])
            PrimitiveListPage._s_booleansSpec = PropSpec("booleans", None, [list, bool, ])
            PrimitiveListPage._s_intsSpec = PropSpec("ints", None, [list, int, ])
            PrimitiveListPage._s_floatsSpec = PropSpec("floats", None, [list, float, ])

        self._strings = Prop(self, PrimitiveListPage._s_stringsSpec)
        self._booleans = Prop(self, PrimitiveListPage._s_booleansSpec)
        self._ints = Prop(self, PrimitiveListPage._s_intsSpec)
        self._floats = Prop(self, PrimitiveListPage._s_floatsSpec)
# GENERATED CONSTRUCTOR END

# GENERATED PROPS START
    @property
    def strings(self):
        return self._strings.value

    @property
    def booleans(self):
        return self._booleans.value

    @property
    def ints(self):
        return self._ints.value

    @property
    def floats(self):
        return self._floats.value

    @property
    def props(self):
        return super(PrimitiveListPage, self).props + [self._strings, self._booleans, self._ints, self._floats, ]
# GENERATED PROPS END

File: runtimes/objc/Tests/PrimitiveListPage.m

#import "PrimitiveListPage.h"
#import "microtome.h"

static MTPropSpec* s_stringsSpec = nil;
static MTPropSpec* s_booleansSpec = nil;
static MTPropSpec* s_intsSpec = nil;
static MTPropSpec* s_floatsSpec = nil;

@implementation PrimitiveListPage {
@protected
    MTObjectProp* _strings;
    MTObjectProp* _booleans;
    MTObjectProp* _ints;
    MTObjectProp* _floats;
}

- (NSArray*)strings { return _strings.value; }
- (NSArray*)booleans { return _booleans.value; }
- (NSArray*)ints { return _ints.value; }
- (NSArray*)floats { return _floats.value; }

- (NSArray*)props { return MT_PROPS(_strings, _booleans, _ints, _floats, ); }

- (id)init {
    if ((self = [super init])) {
        _strings = [[MTObjectProp alloc] initWithPropSpec:s_stringsSpec];
        _booleans = [[MTObjectProp alloc] initWithPropSpec:s_booleansSpec];
        _ints = [[MTObjectProp alloc] initWithPropSpec:s_intsSpec];
        _floats = [[MTObjectProp alloc] initWithPropSpec:s_floatsSpec];
    }
    return self;
}

+ (void)initialize {
    if (self == [PrimitiveListPage class]) {
        s_stringsSpec = [[MTPropSpec alloc] initWithName:@"strings"
            annotations:nil
            valueClasses:@[ [NSArray class],[NSString class], ] ];
        s_booleansSpec = [[MTPropSpec alloc] initWithName:@"booleans"
            annotations:nil
            valueClasses:@[ [NSArray class],[BOOL class], ] ];
        s_intsSpec = [[MTPropSpec alloc] initWithName:@"ints"
            annotations:nil
            valueClasses:@[ [NSArray class],[int class], ] ];
        s_floatsSpec = [[MTPropSpec alloc] initWithName:@"floats"
            annotations:nil
            valueClasses:@[ [NSArray class],[float class], ] ];
    }
}

@end
